zerver/lib/upload.py

```diff
@@ -671,12 +692,29 @@ def upload_emoji_image(
             user_profile,
             resized_image_data,
         )
+        if is_animated:
+            upload_image_to_s3(
+                self.avatar_bucket,
+                emoji_path.replace(".gif", "-still.png"),
+                "image/png",
+                user_profile,
+                cast(bytes, still_image_data),
```

Comment:
Rather than doing these casts, we should just `assert still_image_data is not None` in this block.

Reply:
Resolved this with my push.